src/pages/ValidateLogin2FANewWorkspacePage.js

```diff
@@ -0,0 +1,146 @@
+            // by calling dismissModal(), the /v/... route is removed from history so the user will get taken to `/`
+            // if they cancel out of the new workspace modal.
+            Navigation.dismissModal();
+            Navigation.navigate(ROUTES.WORKSPACE_NEW);
+        }
+    }
+
+    validateAndSubmitForm() {
+        if (!this.state.twoFactorAuthCode.trim()) {
+            this.setState({formError: this.props.translate('passwordForm.pleaseFillOutAllFields')});
+            return;
+        }
+
+        this.setState({
+            formError: null,
+        });
+
+        const accountID = lodashGet(this.props.route.params, 'accountID', '');
+        const validateCode = lodashGet(this.props.route.params, 'validateCode', '');
+        continueSessionFromECom(accountID, validateCode, this.state.twoFactorAuthCode);
+    }
+
+    render() {
+        // If the user is already logged in, don't need to display anything because they will get redirected to the
+        // new workspace page in componentDidMount
+        if (this.props.session.authToken) {
+            return null;
+        }
+
+        return (
+            <View style={[styles.signInPageInnerNative]}>
+                <View style={[styles.signInPageLogoNative]}>
+                    <ExpensifyCashLogo width={variables.componentSizeLarge} height={variables.componentSizeLarge} />
+                </View>
+                <View style={[styles.mb6, styles.alignItemsCenter]}>
+                    <Text style={[styles.h1]}>
+                        {this.props.translate('signInPage.expensifyDotCash')}
+                    </Text>
+                </View>
+
+                <View style={[styles.signInPageFormContainer]}>
+                    <View style={[styles.mb4]}>
+                        <Text style={[styles.formLabel]}>
+                            {this.props.translate('passwordForm.enterATwoFactorAuthenticationCodeToContinue')}
+                        </Text>
+                        <TextInput
+                            style={[styles.textInput]}
+                            value={this.state.twoFactorAuthCode}
+                            placeholder={this.props.translate('passwordForm.twoFactorCode')}
+                            placeholderTextColor={themeColors.placeholderText}
+                            onChangeText={text => this.setState({twoFactorAuthCode: text})}
+                            onSubmitEditing={this.validateAndSubmitForm}
+                            keyboardType={CONST.KEYBOARD_TYPE.NUMERIC}
+                        />
+                    </View>
+                    <View>
+                        <Button
+                            success
+                            style={[styles.mb2]}
+                            text={this.props.translate('common.continue')}
+                            isLoading={this.state.loading}
```

Comment:
NAB: We never set this anywhere but the constructor, meaning the user has no feedback as to when we are actually loading

Reply:
I'll see about cleaning this up.